### PR TITLE
cpe: add additional handling for invalid URI inputs

### PR DESCRIPTION
**Pull request by hdonnay**
Found a case where invalid input resulted in the component being interpreted incorrect during a match. This should fix that bug.

### CODE REVIEW

**Comment by hdonnay** [comment]
The alternative to this sort of escaping is to report an error. But since we're accepting these as input currently, this might be the least bad solution.

Another fix would be to replace the `strings.Replacer` with some code that explicitly does a normalization pass, percent-encoding things that should be escaped and decoding things that shouldn't.

**Comment by hdonnay** [comment]
After some offline discussion, decided to have these inputs error, instead.

File: toolkit/types/cpe/unbind.go [hunk 1]
```diff
@@ -64,16 +64,24 @@ func UnbindURI(s string) (WFN, error) {
...
+		if err := r.Attr[attrs[i]].unbindURI(&b, c); err != nil {
+			return WFN{}, fmt.Errorf("cpe: %v: %w", attrs[i], err)
+		}
 	}
 	return r, r.Valid()
 }
 
-func (v *Value) unbindURI(b *strings.Builder, s string) {
+func (v *Value) unbindURI(b *strings.Builder, s string) error {
+	// From the characters that should be escaped, Table 6-1.
```
Comment (crozzy) [inline toolkit/types/cpe/unbind.go:81]: I can't find the Table 6-1 (or [here](https://nvlpubs.nist.gov/nistpubs/Legacy/IR/nistir7696.pdf) it seems to show something different), would it be possible to add the link as well?

**Review by crozzy — CHANGES_REQUESTED** [review]
Sounds good, just one comment

File: toolkit/types/cpe/unbind.go (hunk 1 above)
Comment (hdonnay) [reply toolkit/types/cpe/unbind.go:81]: That's the matching specification, not the naming spec. I'll add a link.

**Review by crozzy — APPROVED** [review]

**Pull request merged**